src/transports/smart_protocol.c

```diff
@@ -285,20 +284,26 @@ static int fetch_setup_walk(git_revwalk **out, git_repository *repo)
 	git_revwalk_sorting(walk, GIT_SORT_TIME);
 
 	for (i = 0; i < refs.count; ++i) {
+		git_reference *ref;
+
 		/* No tags */
 		if (!git__prefixcmp(refs.strings[i], GIT_REFS_TAGS_DIR))
-			continue;
+			goto on_continue;
 
 		if ((error = git_reference_lookup(&ref, repo, refs.strings[i])) < 0)
-			goto on_error;
+			goto on_continue;
 
 		if (git_reference_type(ref) == GIT_REF_SYMBOLIC)
-			continue;
+			goto on_continue;
 
 		if ((error = git_revwalk_push(walk, git_reference_target(ref))) < 0)
-			goto on_error;
+			goto on_continue;
 
+on_continue:
```

Comment:
This label name is misleading, as we don't just go here during continue, but for each iteration.

Reply:
Yeah, I had some problems here finding a proper name